packages/@aws-cdk/aws-msk-alpha/lib/cluster.ts

```diff
@@ -481,10 +502,19 @@ export class Cluster extends ClusterBase {
     }
 
     const instanceType = props.instanceType
-      ? this.mskInstanceType(props.instanceType)
-      : this.mskInstanceType(
-        ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.LARGE),
-      );
+      ? props.instanceType
+      : ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.LARGE);
+
+    if (props.storageMode && props.storageMode === StorageMode.TIERED) {
+      if (!props.kafkaVersion.isTieredStorageCompatible()) {
+        throw Error(`To deploy a tiered cluster you must select a compatible Kafka version, got ${props.kafkaVersion.version}`);
+      }
+      if (this.mskInstanceType(instanceType) === this.mskInstanceType(
```

Comment:
Tried a bunch of different if statements here, in line with your suggestions @lpizzinidev, however the comparison of
- `instanceType === ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL)`
- `prop.instanceType === ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL)`
are not raising errors correctly in the tests, which I expected to work. Seems like the strings compare better than the objects, but I am not sure why.

Reply:
I think that the current implementation should be fine (and probably better than my suggestion) 💪